pkg/utils/k8sutil.go

```diff
@@ -843,3 +749,261 @@ func DeleteIngress(client kubernetes.Interface, name, ns string) error {
+	}
+
+	dpm.Spec.Template.Spec.Volumes = append(dpm.Spec.Template.Spec.Volumes, v1.Volume{
+		Name: funcMeta.Name,
+		VolumeSource: v1.VolumeSource{
+			ConfigMap: &v1.ConfigMapVolumeSource{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: funcMeta.Name,
+				},
+			},
+		},
+	})
+	if len(dpm.Spec.Template.Spec.Containers) == 0 {
+		dpm.Spec.Template.Spec.Containers = append(dpm.Spec.Template.Spec.Containers, v1.Container{})
+	}
+	dpm.Spec.Template.Spec.Containers[0].Image = funcMeta.Image
+	dpm.Spec.Template.Spec.Containers[0].Name = funcMeta.Name
+	dpm.Spec.Template.Spec.Containers[0].Ports = append(dpm.Spec.Template.Spec.Containers[0].Ports, v1.ContainerPort{
+		ContainerPort: 8080,
+	})
+	dpm.Spec.Template.Spec.Containers[0].Env = append(dpm.Spec.Template.Spec.Containers[0].Env,
+		v1.EnvVar{
+			Name:  "FUNC_HANDLER",
+			Value: funcMeta.Handler,
+		},
+		v1.EnvVar{
+			Name:  "MOD_NAME",
+			Value: funcMeta.Module,
+		},
+		v1.EnvVar{
+			Name:  "TOPIC_NAME",
+			Value: funcObj.Spec.Topic,
+		})
+	dpm.Spec.Template.Spec.Containers[0].VolumeMounts = append(dpm.Spec.Template.Spec.Containers[0].VolumeMounts, v1.VolumeMount{
+		Name:      funcMeta.Name,
+		MountPath: "/kubeless",
+	})
+
+	// update deployment for custom runtime
+	if funcObj.Spec.Deps != "" {
+		updateDeployment(dpm, funcObj.Spec.Runtime)
+		//TODO: remove this when init containers becomes a stable feature
+		addInitContainerAnnotation(dpm)
+	}
+
+	// add liveness Probe to deployment
+	if funcObj.Spec.Type != pubsubFunc {
+		livenessProbe := &v1.Probe{
+			InitialDelaySeconds: int32(3),
+			PeriodSeconds:       int32(30),
+			Handler: v1.Handler{
+				HTTPGet: &v1.HTTPGetAction{
+					Path: "/healthz",
+					Port: intstr.FromInt(8080),
+				},
+			},
+		}
+		dpm.Spec.Template.Spec.Containers[0].LivenessProbe = livenessProbe
+	}
+
+	_, err = client.Extensions().Deployments(funcMeta.Namespace).Create(dpm)
+	if err != nil && k8sErrors.IsAlreadyExists(err) {
+		data, _ := json.Marshal(dpm)
+		_, err = client.Extensions().Deployments(funcMeta.Namespace).Patch(dpm.Name, types.StrategicMergePatchType, data)
+		if err != nil {
+			return err
+		}
+
+		// kick existing function pods then it will be recreated
+		// with the new data mount from updated configmap.
+		// TODO: This is a workaround.  Do something better.
+		pods, err := GetPodsByLabel(client, funcMeta.Namespace, "function", funcMeta.Name)
+		for _, pod := range pods.Items {
+			err = client.Core().Pods(funcMeta.Namespace).Delete(pod.Name, &metav1.DeleteOptions{})
+			if err != nil && !k8sErrors.IsNotFound(err) {
+				// non-fatal
+				logrus.Warnf("Unable to delete pod %s/%s, may be running stale version of function: %v", funcMeta.Namespace, pod.Name, err)
+			}
+		}
+	}
+
+	return err
+}
+
+func ensureFuncJob(client kubernetes.Interface, funcMeta metadata, funcObj *spec.Function, or []metav1.OwnerReference) error {
+	podAnnotations := map[string]string{
+		// Attempt to attract the attention of prometheus.
```

Comment:
Remove - This container will live so briefly that this won't be useful (and it doesn't export prometheus metrics anyway)